common/interpreter/wawaka_wasm/WasmCryptoExtensions.cpp

```diff
@@ -552,7 +552,82 @@ extern "C" bool crypto_hash_wrapper(
             return false;
 
         ByteArray msg(msg_buffer, msg_buffer + msg_buffer_length);
-        ByteArray hash = pcrypto::ComputeMessageHash(msg);
+        ByteArray hash;
+        pcrypto::SHA256Hash(msg, hash);
+        if (hash.size() == 0)
+            return false;
```

Comment:
Not sure when this size would be 0.
Probably the only thing that could fail is the resize, but that would throw an exception.

Speaking of errors, something we missed is the error check on the SHA functions.
Init, Update and Final should return 1 on success, 0 on error.

Reply:
Good point. I don't think there was ever any error checking on the intermediate operations.

Reply:
and... as far as this one goes, the hash size might be 0 if the underlying hash function fails before the value is allocated.